modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -135,6 +154,7 @@ void enableDisableDiskCache(Boolean isDiskCacheEnabled) {
         // When disk cache is disabled, we are not clearing up the disk cache entries yet as that should be part of
         // separate cache/clear API.
         this.caches.put(diskCache, isDiskCacheEnabled);
+        this.statsHolder.setDiskCacheEnabled(isDiskCacheEnabled);
```

Comment:
Is the diskCacheEnabled dynamic setting? How do we track the miss/hits in ancestor nodes if the diskCache is disabled? Is it just based on the onheap tier?

Reply:
Yeah, as of Sagar's recent PR this is dynamic. In case disk cache is disabled, we increment the ancestor nodes for on-heap tier as well.

In the case that it's on and then turns off, we should get the correct logic too: for example, let's say we had 8 heap misses and 4 disk misses when the disk cache was on. We should only count those 4 disk misses towards the total cache misses. Then we turn the disk cache off and get a further 10 heap misses. These should count towards the total, so the total cache has had 14 misses and the heap tier has had 18 misses. If you aggregate by level = tier, you will see onheap = 18, but if you don't, you will see request_cache = 14.